src/url-getters.cpp

```diff
@@ -26,7 +26,7 @@ namespace ada {
       output += "/.";
     }
 
-    output += get_pathname() + (query.has_value() ? "?" + query.value() : "") + (fragment.has_value() ? "#" + fragment.value() : "");
+    output += get_pathname() + get_search() + get_hash();
```

Comment:
This is not what the standard says.

The href getting does not call the search and hash getters...


It says...

> If query is non-null, then set this’s query object’s list to the result of parsing query.   If  url’s fragment is non-null, then append U+0023 (#), followed by url’s fragment, to output.
           

So this PR would introduce a bug.